onnx/version_converter/adapters/sum_8_7.h

```diff
@@ -0,0 +1,34 @@
+// Adapter for Sum in default domain from version 8 to 7
+
+#pragma once
+
+#include "onnx/version_converter/adapters/adapter.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+class Sum_8_7 final : public Adapter {
+  public:
+    explicit Sum_8_7(): Adapter("Sum", OpSetID(8), OpSetID(7)) {}
+
+    void adapt_sum_8_7(std::shared_ptr<Graph> graph, Node* node) const {
+      // Throw an exception if any broadcasting occurs
+      const ArrayRef<Value*>& inputs = node->inputs();
+      // Determine if inputs are of different sizes
+      for (int i = 1; i < (int) inputs.size(); i++) {
+        std::vector<Dimension> A_sizes = inputs[i - 1]->sizes();
+        std::vector<Dimension> B_sizes = inputs[i]->sizes();
+        std::string error = "Sum in OpSet Version 6 does not support broadcasting";
+        ONNX_ASSERT(A_sizes.size() == B_sizes.size());
+        for (int j = 1; j < (int) A_sizes.size(); j++) {
```

Comment:
Let's use the general `broadcastable_numpy` to replace the adhoc/repeated check in each adapter.

Reply:
May have to change this once the Broadcast Adapters are merged

Reply:
Reshape will be branched from https://github.com/onnx/onnx/pull/1284

Reply:
Addressed in https://github.com/onnx/onnx/pull/1292